### PR TITLE
base: add 'isolated' property

### PR DESCRIPTION
**Pull request by Mintsuki**

### CODE REVIEW

File: xbstrap/base.py [hunk 1]
```diff
@@ -685,6 +685,13 @@ def pkg_dependencies(self):
         if "pkgs_required" in self._this_yml:
             yield from self._this_yml["pkgs_required"]
 
+    @property
+    def isolated(self):
+        if "isolated" not in self._this_yml:
+            return False
+        else:
+            return self._this_yml["isolated"]
```
Comment (ArsenArsen) [inline xbstrap/base.py:693]: `return self._this_yml.get("isolated", False)`

File: xbstrap/base.py [hunk 2]
```diff
@@ -1647,10 +1654,13 @@ def substitute(varname):
 
     # Build the environment
     environ = os.environ.copy()
+    if manifest["isolated"]:
+        environ = {}
```
Comment (ArsenArsen) [inline xbstrap/base.py:1658]: Invert this condition

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
See also: the last comment (accidentally posted individually).

How does this work? I've never seen anyone package host-coreutils. This also is not isolation at all, that'd be done via cbuildrt/docker/whatever. Is there anything that this does and that a container wouldn't?

**Comment by Mintsuki** [comment]
> See also: the last comment (accidentally posted individually).
> 
> How does this work? I've never seen anyone package host-coreutils. This also is not isolation at all, that'd be done via cbuildrt/docker/whatever. Is there anything that this does and that a container wouldn't?

Yeah it's not isolation in that sense, the name isn't something I particularily care about, you can choose whichever other name you prefer.

This option is meant to enforce builds using bootstrap provided host tools, rather than relying on host OS tools whatsoever, in order to improve reproducibility and consistency of builds across different host OSes without needing containers.

**Pull request closed without merging**